mypy/semanal.py

```diff
@@ -719,7 +721,8 @@ def unbind_class_type_vars(self) -> None:
     def analyze_class_decorator(self, defn: ClassDef, decorator: Expression) -> None:
         decorator.accept(self)
 
-    def calculate_abstract_status(self, typ: TypeInfo) -> None:
+    @staticmethod
+    def calculate_abstract_status(typ: TypeInfo) -> None:
```

Comment:
By convention, we usually prefer module-level methods to static methods in the mypy codebase. This would probably be better as a module-level method. There are exceptions such as helper methods that don't make sense outside a particular class, or things that are tightly coupled with the implementations details of a class -- they may be fine as static methods. By default, we use a module-level method if we don't need `self`.

Rationale:
* Module-level methods make it obvious on the call site that the method doesn't use `self`, and without having to use the class prefix.
* Module-level methods are easier to move to a different module and reuse, and generally make it a bit easier to refactor code.
* That's how most of the rest of the codebase is written, and consistency is nice.